### DISCUSSION TITLE
Difference between biometric live capture and scheduled options

### DISCUSSION THREAD

**Discussion by vanyell** [horilla-opensource/horilla#300] [General]
### Need some clarity on the scope of time being retrieved by those two options.

Live capture - is the of retrieval of the attendance records starting from the time  option being enabled?
Is this the same with the scheduled option?

E.g.

Live capture or schedule capture was enabled at 8:24am. Hence only subsquent records after 8:24am are processed for capture into the system. 

### Which option is recommended.

Currently doing biometric live capture. However this option seems to turn off by itself overnight (don't have night shift). Is there a builtin timeout?

**Comment by horilla-opensource** [comment]
Hi @vanyell ,

1. **Live Capture**  : When enabled, Horilla immediately starts retrieving attendance records from the biometric device. It captures data every second using a thread. However, live capture may automatically turn off due to an inherent timeout, inactivity, server restarts, or network issues. 

2. **Scheduled Capture**  : Similar to live capture, scheduled capture retrieves attendance records from the biometric device but follows a user-defined schedule. For example, if set to run every minute, it will fetch data at intervals like 08:24 AM, then 08:25 AM, with up to a one-minute delay. While live capture retrieves data every second, if it turns off, any attendance records during that downtime will not be captured. However, the scheduled capture will still fetch records based on the last successful retrieval time.

**Reply by vanyell** [reply]
With that explanation, going for scheduled capture is recommended than using live capture then...

**Reply by horilla-opensource** [reply]
Yes, we recommend the scheduled capture.

**Reply by vanyell** [reply]
Have a followup question. Since the python library retrieves all available records on the device. what would be the recommended schedule to get in/out records? End of the business day? every hour? How does horilla determine in/out? Usually, the first checkin time and last checkout time is considered here.

**Reply by horilla-opensource** [reply]
Horilla determines whether an attendance activity is a "clock-in" or "clock-out" based on the biometric punch code provided by the ZK device. 

Specifically:
**Clock-In**: When the punch code is one of {0, 3, 4}, Horilla interprets the action as a "clock-in." These punch codes typically represent different forms of check-ins (e.g., starting work or returning from a break).

**Clock-Out**: For any other punch codes not in {0, 3, 4}, Horilla interprets the activity as a "clock-out," meaning the employee is leaving or ending work.
